src/v/serde/test/serde_test.cc

```diff
@@ -657,6 +678,7 @@ struct new_cs
     bool operator==(const new_cs&) const = default;
 
     std::vector<test_msg1_new_manual> data_;
+    auto serde_fields() { return std::tie(data_); }
```

Comment:
so this is the case where writes go to serde_fields, but reads go through read_nested?

Reply:
yeah, there may be others in this file. i've seen it in actual live code as well, but an example hasn't sprung to mind.

Reply:
you mean like xor, right? as in disallow cases like this one?

Reply:
yeh, xor. i guess we talked about that before. nm

Reply:
all good. i agree - will need another round of updates & reviews, so no harm in punting to follow up.